src/harness/testcases/spectrum_inquiry_testcase.py

```diff
@@ -391,6 +391,95 @@ def test_WINNF_FT_S_SIQ_16(self):
       self.assertEqual(resp['response']['responseCode'], 0)
 
   @winnforum_testcase
+  def test_WINNF_FT_S_SIQ_17(self):
+    """This test ensures that when a message with two requests, 
+    both for PAL channels, is sent with valid parameters, 
+    the SAS sends a successful response for both the requests.
+    Response Code must be 0 for all CBSDs
+    """
+
+    # Load the Data
+    pal_low_frequency = 3550000000.0
+    pal_high_frequency = 3560000000.0
+    registration_request = []
+    ppa_records = []
+    pal_records = []
+    for device_filename, pal_filename, ppa_filename in zip(
+      ('device_a.json', 'device_c.json'),
+      ('pal_record_0.json', 'pal_record_1.json'),
+      ('ppa_record_0.json', 'ppa_record_1.json')):
+
+      device = json.load(
+        open(os.path.join('testcases', 'testdata', device_filename)))
+      pal_record = json.load(
+        open(os.path.join('testcases', 'testdata', pal_filename)))
+      ppa_record = json.load(
+        open(os.path.join('testcases', 'testdata', ppa_filename)))
+      ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                              [pal_record],
+                                                              pal_low_frequency,
+                                                              pal_high_frequency,
+                                                              device['userId'])
+
+      # Move the Device to a random location in PPA
+      device['installationParam']['latitude'], \
+      device['installationParam']['longitude'] = getRandomLatLongInPolygon(ppa_record)
+      ppa_records.append(ppa_record)
+      pal_records.append(pal_record[0])
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+      registration_request.append(device)
+
+    # Register the devices
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      cbsd_ids.append(resp['cbsdId'])
+    del request, response
+
+    for ppa_record, pal_record, cbsd_id in zip(ppa_records, pal_records, cbsd_ids):
+      # Update PPA Record with CBSD ID and Inject Data
+      ppa_record['ppaInfo']['cbsdReferenceId'] = [cbsd_id]
```

Comment:
The specs do not allow exchange of CBSD IDs directly because (for reasons I don't know) we are not supposed to exchange CBSD serial numbers with other SASes. So the CBSD Reference ID must be as described in the SAS-SAS TS when we are exchanging information between SASes (i.e. there is no potential to change the SAS-SAS spec at this time) and it is not necessarily the same as the CBSD ID (technically a SAS could choose to create the CBSD ID using the same methodology, but there is no requirement).

tl;dr: already there's a difference between the CBSD ID in the SAS-CBSD exchange and the CBSD Reference ID in the SAS-SAS exchange (and we can't change either of those decisions).

For the testing API *only*, I think it makes sense to use the CBSD ID instead of the CBSD Reference ID. If we use the CBSD Reference ID, the SAS would need to maintain mapping information so it can immediately translate the CBSD Reference ID to the CBSD ID (not a simple function since SHA-1 is not reversible). There's no other part of the SAS that requires this mapping information/functionality so we would add it only for testing purposes, which is something I generally oppose :) If we use the CBSD ID, the SAS can use the ID directly -- the same as it would do in normal SAS-SAS exchange.

I do agree that if we decide to use the CBSD ID in this case that we should document it clearly so that there's no confusion in the future.